### PR TITLE
Behat extension to resize selenium window

### PR DESCRIPTION
**Pull request by craig410**

### CODE REVIEW

**Comment by craig410** [comment]
@acoulton bit of a first stab in the dark at a behat extension but appears to work when dropped into one of the projects and based entirely on the existing extensions. Thoughts welcome :)

File: src/Extension/ResizeBrowserWindowExtension.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * @author    Craig Gosman <craig@ingenerator.com>
+ * @licence   proprietary
```
Comment (acoulton) [inline src/Extension/ResizeBrowserWindowExtension.php:4]: Should be `@license BSD-3-Clause` (or omitted) to match the overall package license. Although I see my IDE settings have put proprietary on all the other files here :)

File: src/Extension/ResizeBrowserWindowExtension/ResizeBrowserWindowListener.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * Hook fired before every scenario that starts (or checks status of) phantomjs when first required
+     *
+     * @param BeforeScenarioTested $event
+     */
+    public function beforeScenario(BeforeScenarioTested $event)
+    {
+        if ( ! $event->getScenario()->hasTag('javascript')) {
+            return;
+        }
+
```
Comment (acoulton) [inline src/Extension/ResizeBrowserWindowExtension/ResizeBrowserWindowListener.php:58]: Probably best to add a state flag here to return if this has already run (so it only actually does anything on the first JS scenario in a suite)

ISTR at the moment calling start more than once isn't an issue but there's potential risk of future weirdness / conflicts with other stuff.

**Review by acoulton — APPROVED** [review]
Awesome! Couple of minor things but other than that pop it in the changelog and I'd say it's done :)

File: src/Extension/ResizeBrowserWindowExtension/ResizeBrowserWindowListener.php (hunk 2 above)
Comment (craig410) [reply src/Extension/ResizeBrowserWindowExtension/ResizeBrowserWindowListener.php:58]: Added a state flag to only run it on the first execution. Was wrongly under the impression there was a new session per scenario.

File: src/Extension/ResizeBrowserWindowExtension.php (hunk 1 above)
Comment (craig410) [reply src/Extension/ResizeBrowserWindowExtension.php:4]: Hadn't spotted it either, mine is doing the same. Could address this in another pull to change the lot as it looks like its everywhere?

Comment (acoulton) [reply src/Extension/ResizeBrowserWindowExtension.php:4]: Yeah good plan. I think best best is just leave the license/licence tag out.

**Pull request merged**